### ISSUE TITLE
add: remove_sonos_playlist

### ISSUE THREAD

**Issue by dundeemt** [SoCo/SoCo#341]
There are currently a couple of ways to create a sonos playlist but no way to delete them.  I am getting ready to do a PR to add this functionality.  I do have a couple questions and figured it best to get them sorted here before I create the PR.

1) parameter name - the underlying contentDirectory.DestroyObject call requires an 'object_id' to be set. I have the function signature looking like this: `remove_sonos_playlist(self, object_id)`  The content of object_id should be the be the item_id (i.e. SQ:10) of the playlist to delete.  object_id and item_id are used in the code and I am not sure if the signature should be object_id or item_id.  I guessed object_id since that is what the underlying call is using.

2) Proper position in the SoCo class doc string for the new function.  Methods and Properties are split out but within those subgroups it is not strictly alpha order.  It appears to be sub-grouped by functionality and then alpha within that.  I have it listed after add_item_to_sonos_playlist in the docstring and within the code.  Ok? OR Should it come after the two create_... methods?

3) Testing - against the mock soco object only or against the mock and an actual device (--ip)? I am unsure if the project is trying to go completely mock or if the mock usage is just to augment testing when you don't have a reachable device.

4) doc strings and docs - I see a variety of doc string/documenting methods in play.  Which is the preferred version?  The method I used for to_dict @KennethNielsen behaved rather badly when my parameter doc string needed to wrap and I had to butcher it to make it fit on one line within the length limit of pyflake/pep8. I am pretty sure it was my formatting or some such, but I was unsure of where to read up on the requirements to make the output work as intended. 

Best,
Jeff

**Comment by lawrenceakka** [comment]
Hi Jeff,

> 1) parameter name - the underlying contentDirectory.DestroyObject call requires an 'object_id' to be set. I have the function signature looking like this: remove_sonos_playlist(self, object_id) The content of object_id should be the be the item_id (i.e. SQ:10) of the playlist to delete. object_id and item_id are used in the code and I am not sure if the signature should be object_id or item_id. I guessed object_id since that is what the underlying call is using.
> 

item_id seems to be what Sonos uses to refer to the id of a metadata or music item (e.g. a track or album). It uses object_id for UPnP calls.  So I guess object_id is correct here.


> 2) Proper position in the SoCo class doc string for the new function. Methods and Properties are split out but within those subgroups it is not strictly alpha order. It appears to be sub-grouped by functionality and then alpha within that. I have it listed after add_item_to_sonos_playlist in the docstring and within the code. Ok? OR Should it come after the two create_... methods?
> 

I am not sure that people have paid that much attention to it, and the whole docstring probably needs rewriting anyway. I have tried to keep related functions together (e.g. all transport functions, all queue functions etc). Put it where you have suggested.  At some point, we may split `core.py` into several smaller files.



> 3) Testing - against the mock soco object only or against the mock and an actual device (--ip)? I am unsure if the project is trying to go completely mock or if the mock usage is just to augment testing when you don't have a reachable device.
> 

Originally, all testing was done 'live'. But we then introduced unittests and mocks, and wrote the --ip flag. I tend to test against mocks only most of the time, and occasionally against a live device. It would be nice to have both for the functionality you are suggesting.  

> 4) doc strings and docs - I see a variety of doc string/documenting methods in play. Which is the preferred version? The method I used for to_dict @KennethNielsen behaved rather badly when my parameter doc string needed to wrap and I had to butcher it to make it fit on one line within the length limit of pyflake/pep8. I am pretty sure it was my formatting or some such, but I was unsure of where to read up on the requirements to make the output work as intended.

Not sure what problems you have, but we are now trying to use Google style docstrings (though we are not very good at it!)  See here: http://sphinxcontrib-napoleon.readthedocs.org/en/latest/example_google.html  Some of the older code is not updated yet.

Lawrence

**Comment by dundeemt** [comment]
attempting test membership of a DIdlPlayListContainer to the items in .get_sonos_playlists() always fails.
```python
new_playlist = soco.create_new_playlist(title='foo')
assert new_playlist in soco.get_sonos_playlists()     # fails
assert new_playlist.item_id in [x.item_id for x in soco.get_sonos_playlists()]   # passes
```
Is this because the items from the .get_sonos_playlists are not considered playable?  see soco/data_structures.py::DidlObject.\__eq__

Currently, I am comparing using .item_id to test for membership. However, I was confused and a bit disappointed that I couldn't use the first assert.

**Comment by KennethNielsen** [comment]
Hi @dundeemt 

> Is this because the items from the .get_sonos_playlists are not considered playable?

Well, they are playable and therefore certainly also should be considered as such.

Can you check what actually differs? ``__eq__`` checks for equality of the to_dict values, so it would be nice to what actually fails.

**Comment by dundeemt** [comment]
The following code:
```python
created_pl = soco.create_sonos_playlist(title=self.pl_name)
print created_pl.to_dict()
for pl in soco.get_sonos_playlists():
    print pl.to_dict()
assert created_pl.to_dict() in [x.to_dict() for x in soco.get_sonos_playlists()]
```
emits the following:
```python
{u'title': u'zSocoTestPlayList42', u'restricted': True, u'parent_id': u'SQ:', u'item_id': 'SQ:118', u'resources': [<DidlResource 'file:///jffs/settings/savedqueues.rsq#118' at 0x7f9e4890cb50>], u'desc': u'RINCON_AssociatedZPUDN'}
... snipped ...
{u'title': u'zSocoTestPlayList42', u'restricted': True, u'parent_id': u'SQ:', u'item_id': u'SQ:118', u'resources': [<DidlResource 'file:///jffs/settings/savedqueues.rsq#118' at 0x7f9e4890c810>], u'desc': None}
```
The DidlPlaylistContainer emitted by `create_sonos_playlist` has differences starting at `item_id`, the returned value is a 'SQ:118' vs. u'SQ:118' .

The `resources` element has not been converted to a dictionary and it should be if it inherits from DidlObject -- correct? Or it is overridden somewhere. So  comparing the \__repr__ of the objects fails because of location differences.

And finally, the `desc` element has a value, and none of the results from `.get_sonos_playlists` do.

**Comment by dundeemt** [comment]
closed by PR #345
